File: Serene/Serene.Script/Imports/ServerImports/Northwind.SalesByCategoryService.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.Northwind
{
    [Imported, PreserveMemberCase]
    public partial class SalesByCategoryService
    {
        [InlineConstant] public const string BaseUrl = "Northwind/SalesByCategory";

        [InlineCode("Q.serviceRequest('Northwind/SalesByCategory/List', {request}, {onSuccess}, {options})")]
        public static jQueryXmlHttpRequest List(ListRequest request, Action<ListResponse<SalesByCategoryRow>> onSuccess = null, ServiceCallOptions options = null)
        {
            return null;
        }

        [Imported, PreserveMemberCase]
        public static class Methods
        {
            [InlineConstant] public const string List = "Northwind/SalesByCategory/List";
        }
    }
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Northwind.SalesByCategoryService.ts
namespace Serene.Northwind {
    export namespace SalesByCategoryService {
        export const baseUrl = 'Northwind/SalesByCategory';

        export declare function List(request: Serenity.ListRequest, onSuccess?: (response: Serenity.ListResponse<SalesByCategoryRow>) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;

        export namespace Methods {
            export declare const List: string;
        }

        ['List'].forEach(x => {
            (<any>SalesByCategoryService)[x] = function (r, s, o) { return Q.serviceRequest(baseUrl + '/' + x, r, s, o); };
            (<any>Methods)[x] = baseUrl + '/' + x;
        });
    }
}

